consumer/consumer.go

```diff
@@ -104,6 +104,44 @@ func New(config CompletedConfig, client kessel.ClientProvider, logger *log.Helpe
 	}, nil
 }
 
+// NewWithConsumer instantiates a new InventoryConsumer with a provided consumer (useful for testing)
+func NewWithConsumer(config CompletedConfig, client kessel.ClientProvider, logger *log.Helper, consumer Consumer) (InventoryConsumer, error) {
```

Comment:
I wonder if its worth updating the existing New function to take an optional consumer and we check if consumer is nil or not. If its nil, we setup the consumer like we do now using `consumer, err := kafka.NewConsumer(config.KafkaConfig)`. If its not, we just move along and use the provided consumer in the return. Not a hard requirement but an idea.

This would at least prevent duplicating the other code since, unless im missing it, is identical.

Reply:
Going to take this and throw it in Inventory API too since it has the same test issue of trying to connect to Kafka. Thanks!